utils/merkletrie/filesystem/node_test.go

```diff
@@ -196,6 +199,27 @@ func (s *NoderSuite) TestDiffDirectory(c *C) {
 	c.Assert(a, Equals, merkletrie.Modify)
 }
 
+func (s *NoderSuite) TestSocket(c *C) {
+	td, err := os.MkdirTemp("", "socket-test")
+	c.Assert(err, IsNil)
+
+	sock, err := net.ListenUnix("unix", &net.UnixAddr{Name: fmt.Sprintf("%s/socket", td), Net: "unix"})
+	c.Assert(err, IsNil)
+
+	defer sock.Close()
+
+	fsA := osfs.New(td)
+	WriteFile(fsA, "foo", []byte("foo"), 0644)
+
+	noder := NewRootNode(fsA, nil)
+	childs, err := noder.Children()
+	c.Assert(err, IsNil)
+
+	c.Assert(childs, HasLen, 1)
+
+	defer os.RemoveAll(td)
```

Comment:
Could you move this to the top after `MkdirTemp`

Reply:
yepp done!